src/source/view/controls/TextInput.bs

```diff
@@ -0,0 +1,207 @@
+import "pkg:/source/core/Utils.bs"
+import "pkg:/source/view/BaseView.bs"
+
+namespace mv.controls
+  @observersWaitInitialize
+  @node("mv_TextInput", "mv_BaseView")
+  class TextInput extends mv.BaseView
+
+    '+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+    '++ fields
+    '+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+    ' styles are expected to have json for states: (normal|focused|error|disabled)
+    @alwaysnotify
+    @observer("onStyleKeyChange")
+    public styleKey = "mv.TextInput.default"
+
+    @debounce
+    @alwaysnotify
+    @observer("redraw")
+    public text = ""
+
+    @debounce
+    @alwaysnotify
+    @observer("redraw")
+    public hintText = ""
+
+    @debounce
+    @alwaysnotify
+    @observer("redraw")
+    public subText = ""
+
+    @debounce
+    @alwaysnotify
+    @observer("redraw")
+    public errorText = ""
+
+    @debounce
+    @alwaysnotify
+    @observer("redraw")
+    public isFocused = false
+
+    @debounce
+    @alwaysnotify
+    @observer("redraw")
+    public isDisabled = false
+
+    @debounce
+    @alwaysnotify
+    @observer("redraw")
+    public hasError = false
+
+    @debounce
+    @alwaysnotify
+    @observer("redraw")
+    public size = [400, 50]
+
+    @debounce
+    @alwaysnotify
+    @observer("validate")
+    public updatedText = ""
```

Comment:
can we just use text - I find having 2 vars confusing - I don't want validation done _in the textinput_